contrib/inventory/lxc.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+#
+# (c) 2015, Florian Haas, hastexo Professional Services GmbH
+# <florian@hastexo.com>
+# Based in part on:
+# libvirt_lxc.py, (c) 2013, Michael Scherer <misc@zarb.org>
+#
+# This file is part of Ansible,
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Ansible inventory script for LXC containers. Requires Python
+bindings for LXC API.
+
+In LXC, containers can be grouped by setting the lxc.group option,
+which may be found more than once in a container's
+configuration. So, we enumerate all containers, fetch their list
+of groups, and then build the dictionary in the way Ansible expects
+it.
+"""
+
+from __future__ import print_function
+
+# Horrible hack: we want this script to be called lxc.py, yet need to
+# import a module named lxc. Absolute import (from __future__ import
+# absolute_import) does not work here, because this is a script, not a
+# package. The only way to tell Python that "import lxc" does *not*
+# mean "import yourself" is to pop the first entry off of sys.path.
+#
+# Since inventory scripts are always invoked directly, and never
+# imported, we should be able to get away with this.
+import sys
+sys.path.pop(0)
```

Comment:
This hack prevents calling as ```./lxc.py ```:

```
Traceback (most recent call last):
  File "./lxc.py", line 79, in <module>
    main(sys.argv)
  File "./lxc.py", line 70, in main
    result = build_dict()
  File "./lxc.py", line 59, in build_dict
    for c in _lxc.list_containers()])
AttributeError: 'module' object has no attribute 'list_containers'
```

Instead of this, rename the script to ```lxc_inventory.py```.